File: Herencia/Mascota.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Mascota.cpp
 * Author: Usuario
 * 
 * Created on 4 de octubre de 2017, 23:18
 */

#include "Mascota.h"
#include <string>
#include <iostream>
#include <cstdlib>
using namespace std;
Mascota::Mascota() {
}

Mascota::Mascota(string nombre) {

    this->nombre= nombre;
}



string Mascota::getNombre() {
    return this->nombre;
}


void Mascota::setNombre(string nombre) {
    this->nombre=nombre;
}





Mascota::~Mascota() {
}

File: ayudantia7_2019/src/ayudantia7_2019/Mascota.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ayudantia7_2019;

/**
 *
 * @author Juan Pablo Martinez
 */
public abstract class Mascota {
    private String nombre;
    private int peso;

    public Mascota(String nombre, int peso) {
        this.nombre = nombre;
        this.peso = peso;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getPeso() {
        return peso;
    }

    public void setPeso(int peso) {
        this.peso = peso;
    }
    
    abstract public int calcularAlimento();
}
